mautrix_facebook/portal.py

```diff
@@ -573,7 +574,22 @@ async def handle_matrix_message(self, sender: 'u.User', message: MessageEventCon
                                     event_id: EventID) -> None:
         # TODO handle errors?
         # try:
-        await self._handle_matrix_message(sender, message, event_id)
+        try:
+            await self._handle_matrix_message(sender, message, event_id)
+        except Exception as e:
+            sender.send_remote_checkpoint(
+                MessageSendCheckpointStatus.PERM_FAILURE,
+                event_id,
+                self.mxid,
+                EventType.ROOM_MESSAGE,
+                message.msgtype,
+                error=e,
+            )
+            await self._send_bridge_error(str(e))
+            self.log.exception("Failed to handle matrix message.", e)
+            raise
```

Comment:
This probably doesn't need to be raised further if we already log and checkpoint it